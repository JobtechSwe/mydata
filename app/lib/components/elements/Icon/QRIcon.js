import React from 'react'
import { Image, View } from 'react-native'
import styled from '../../../theme'

const StyledQRIcon = styled(View)`
  align-items: center;
  width: 96px;
  height: 96px;
  justify-content: center;
  border-radius: 96px;
  background-color: ${({ theme }) => theme.colors.primary};
`

const StyledQRImage = styled(Image)`
  width: 28px;
  height: 28px;
`

export default QRIcon = () => (
  <StyledQRIcon>
    <StyledQRImage source={require('./img/qr_inactive.png')} />
  </StyledQRIcon>
)
