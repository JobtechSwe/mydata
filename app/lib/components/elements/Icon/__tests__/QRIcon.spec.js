import React from 'react'
import { shallow } from 'react-native-testing-library'
import MockedProvider from '../../../../utils/test-utils'
import QRIcon from '../QRIcon'

describe('elements/Icons/QRIcon', () => {
  it('should render', () => {
    const { output } = shallow(
      <MockedProvider>
        <QRIcon />
      </MockedProvider>
    )

    expect(output).toMatchSnapshot()
  })
})
