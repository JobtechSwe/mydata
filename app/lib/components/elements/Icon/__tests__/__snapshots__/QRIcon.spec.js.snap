// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/Icons/QRIcon should render 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "inactive": "#D8D5F6",
        "lightGrey": "#EAEAF1",
        "primary": "#241C7D",
        "quiet": "#F9F9FB",
        "shadow": "#B1ACEC",
        "white": "#FFFFFF",
      },
      "fontSize": Object {
        "large": "48px",
        "regular": "18px",
      },
      "typography": Object {
        "bold": "Karla-Bold",
        "contrast": "Rubik-Medium",
        "regular": "Karla-Regular",
      },
    }
  }
>
  <QRIcon />
</ThemeProvider>
`;
